google/cloud/aiplatform/metadata/resource.py

```diff
@@ -38,7 +38,8 @@ class _Resource(base.AiPlatformResourceNounWithFutureManager, abc.ABC):
 
     def __init__(
         self,
-        resource_name: str,
+        resource_name: str = None,
```

Comment:
```suggestion
        resource_name: str = Optional[None],
```

Reply:
Should be "resource_name: Optional[str] = None," right?